src/crds_gossip.rs

```diff
@@ -92,9 +94,10 @@ impl CrdsGossip {
 
     /// refresh the push active set
     /// * ratio - number of actives to rotate
-    pub fn refresh_push_active_set(&mut self) {
+    pub fn refresh_push_active_set(&mut self, bank: Option<&Arc<Bank>>) {
```

Comment:
I don't care for all the `Option<Arc<Bank>>` params. How about passing in `&[(Pubkey, u64)]`?

Reply:
I'd have to iter over all the bank accounts and get manually construct that slice. Also lookups will be painful unless I put it into a map and everytime something changes I'll have to reconstruct this. The option exists for when you don't care about having things being stake weighted.

Reply:
@garious I'll see if I can switch it.